cargo-dist/src/lib.rs

```diff
@@ -57,8 +57,90 @@ pub mod tasks;
 #[cfg(test)]
 mod tests;
 
+/// dist env test -- make sure we have everything we need for a build.
+pub fn do_env_test(cfg: &Config) -> DistResult<()> {
+    let (dist, _manifest) = tasks::gather_work(cfg)?;
+
+    let builds = dist.config.builds;
+
+    let need_cargo_auditable = builds.cargo.cargo_auditable;
+    let need_cargo_cyclonedx = builds.cargo.cargo_cyclonedx;
+    let need_omnibor = builds.omnibor;
+    let mut need_xwin = false;
+    let mut need_zigbuild = false;
+
+    let tools = dist.tools;
+
+    let cargo = tools.cargo()?;
```

Comment:
We can't return an error here - we might be building non-Cargo software, and it's valid for users not to have Cargo if they're not building any Rust software.

Instead, we could consider adding cargo to the list of needed tools below, and checking it if we have any `BuildStep::Cargo` jobs. (Can still call `let cargo =` here, but defer checking if we got `Ok` or `Err` until we're inside `BuildStep::Cargo`.)

Reply:
I couldn't find a way to hoist `let cargo = tools.cargo()` out of the loop, but keep the check in the loop, without either ownership problems or adding more code than felt reasonable.

For now, I just moved both the `let cargo = ...` and `let host = ...` lines into the loop. If you want, I can look into having it check for Cargo tomorrow.